kv/org.go

```diff
@@ -246,11 +246,25 @@ func (s *Service) FindOrganizations(ctx context.Context, filter influxdb.Organiz
 		return os, len(os), nil
 	}
 
+	var offset, limit, count int
```

Comment:
I still think you should implement the new options for the above organization slice, maybe something like:
```diff
@@ -1,12 +1,28 @@
+  var offset, limit, count int
+  var descending bool
+  if len(opt) > 0 {
+    // these checks should be part of DecodeFindOffset
+    if opt[0].Offset > 0 {
+      offset = opt[0].Offset
+    }
+
+    if opt[0].Limit > 0 {
+      limit = opt[0].Limit
+    }
+
+    descending = opt[0].Descending
+  }
+
   if filter.UserID != nil {
     // find urms for orgs with this user
     urms, _, err := s.FindUserResourceMappings(ctx, influxdb.UserResourceMappingFilter{
       UserID:       *filter.UserID,
       ResourceType: influxdb.OrgsResourceType,
-    }, opt...)
+    })
     if err != nil {
       return nil, 0, err
     }
+
     // find orgs by the urm's resource ids.
     for _, urm := range urms {
       o, err := s.FindOrganizationByID(ctx, urm.ResourceID)
@@ -15,5 +31,18 @@
         os = append(os, o)
       }
     }
+
+    if offset > 0 && offset < len(os) {
+      os = os[offset:]
+    }
+
+    if limit > 0 && limit < len(os) {
+      os = os[:limit]
+    }
+
+    if descending {
+      os = // reverse `os` how you wish
+    }
+
     return os, len(os), nil
   }
```

Reply:
Yes, you are right, it makes code clean. 
However, (again) I'm following similar duplicate snippets in some other places, so I guess it would be better to start a new refactor-issue to solve all these duplicate snippets?

https://github.com/influxdata/influxdb/blob/ce443f832e87484582eb3d5c78c812c683299a06/kv/notification_rule.go#L370-L391

https://github.com/influxdata/influxdb/blob/ce443f832e87484582eb3d5c78c812c683299a06/kv/dashboard.go#L231-L237

https://github.com/influxdata/influxdb/blob/ce443f832e87484582eb3d5c78c812c683299a06/kv/bucket.go#L406-L412

Reply:
You're doing great. For now I think go ahead and implement the `offset` and `limit` options for the user resource mappings wherever you prefer and this pr should be good to go. Without some implementation for the urm's i don't think the different behavior of using options with/without filters makes for a good user experience.

Reply:
Thanks for your advice! Done.

Reply:
BTW, similar issue in https://github.com/influxdata/influxdb/pull/19330.
(and maybe some other places about `FindOptions`)

Reply:
What do you think? @glinton